src/GraphQL/Validation/DocumentValidator.ParseArgumentVisitor.cs

```diff
@@ -17,12 +17,47 @@ private class ParseArgumentVisitor : ASTVisitor<ParseArgumentVisitor.Context>
 
         private ParseArgumentVisitor() { }
 
+        public override ValueTask VisitAsync(ASTNode? node, Context context)
+        {
+            if (node is IHasDirectivesNode hasDirectivesNode)
+            {
+                // if any directives were supplied in the document for the field or fragment spread,
+                // load all defined arguments for directives on the field or fragment spread
+                if (hasDirectivesNode.Directives?.Count > 0)
+                {
+                    try
+                    {
+                        var directives = ExecutionHelper.GetDirectives(hasDirectivesNode, context.Variables, context.Schema);
+                        if (directives != null)
+                        {
+                            (context.DirectiveValues ??= new()).Add(node, directives);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        // todo: report error properly
+                        context.ValidationContext.ReportError(new ValidationError($"Error trying to resolve {NodeDescription()}.", ex));
+
+                        string NodeDescription() => node switch
+                        {
+                            GraphQLField field => $"field '{field.Name.Value}'",
+                            GraphQLFragmentSpread fragment => $"fragment spread",
+                            GraphQLVariableDefinition varDef => $"variable definition '{varDef.Variable.Name}'",
+                            _ => "node"
+                        };
+                    }
+                }
+            }
+
+            return base.VisitAsync(node, context);
+        }
+
         protected override ValueTask VisitDocumentAsync(GraphQLDocument document, Context context)
         {
             // parses the selected operation only
             var selectedOperation = context.ValidationContext.Operation;
             if (selectedOperation != null)
-                return VisitOperationDefinitionAsync(selectedOperation, context);
+                return VisitAsync(selectedOperation, context);
```

Comment:
Yeah, double dispatch is important here. I would recommend to double check visitor to be sure it calls `VisitAsync` instead of concrete `VisitBlaBlaAsync`.

Reply:
Yes, the parser project's visitor methods all properly call `VisitAsync`.